### PR TITLE
Add an option to specify a Span Kind via the WithSpan annotation.

### LINKED ISSUE #1303: Add the ability to specify a Span Kind using the WithSpan annotation
**Describe the solution you'd like**
Add an optional parameter to the `WithSpan` annotation that specifies the span Kind to create.

**Describe alternatives you've considered**
Doing this manually is possible, but it would be nice to have it built in to the annotation.

### PR DESCRIPTION
**Pull request by jkwatson**
resolves #1303 for the API. Will open an issue with the instrumentation project when this is merged.

### CODE REVIEW

**Comment by trask** [comment]
I think I'd prefer for the annotations package to not have any dependencies, and instead have it's own `SpanKind` enum.

@iNikem @tylerbenson wdyt?

**Comment by iNikem** [comment]
I don't see any downsides in depending on API module. And duplicating means syncing, I don't think it is worth the effort.

**Comment by jkwatson** [comment]
@trask  What about specifically excluding grpc-context from the dependencies of the annotations?

**Comment by jkwatson** [comment]
> @trask What about specifically excluding grpc-context from the dependencies of the annotations?

I added this to the PR.

File: contrib/auto_annotations/build.gradle [hunk 1]
```diff
@@ -9,7 +9,7 @@ description = 'OpenTelemetry Contrib Auto Annotations'
 ext.moduleName = "io.opentelemetry.contrib.auto.annotations"
 
 dependencies {
-    implementation project(':opentelemetry-api')
+    implementation project(':opentelemetry-api') exclude group: 'io.grpc', module: 'grpc-context'
```
Comment (jkwatson) [inline contrib/auto_annotations/build.gradle:12]: @bogdandrutu @carlosalberto any issues with this late-breaking change to minimize the dependencies of the annotations module?

Comment (bogdandrutu) [reply contrib/auto_annotations/build.gradle:12]: First request: Let's do it in a separate PR :)
Second request: I need to understand what you try to achieve :)

Comment (jkwatson) [reply contrib/auto_annotations/build.gradle:12]: 1. sure thing; that's why I asked
2. @trask is concerned that people will need to pull in grpc-context just to use an annotation, and would prefer that not to be the case.

Comment (jkwatson) [reply contrib/auto_annotations/build.gradle]: I removed the change from this PR, FYI

Comment (jkwatson) [reply contrib/auto_annotations/build.gradle]: I removed the change from this PR, FYI

**Pull request merged**